app/src/main/java/org/wikipedia/readinglist/PreviewReadingListRepository.kt

```diff
@@ -0,0 +1,112 @@
+package org.wikipedia.readinglist
+
+import android.content.Context
+import android.util.Base64
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.int
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.dataclient.Service
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.dataclient.mwapi.MwQueryPage
+import org.wikipedia.json.JsonUtil
+import org.wikipedia.readinglist.database.ReadingList
+import org.wikipedia.readinglist.database.ReadingListPage
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.DateUtil
+import org.wikipedia.util.ImageUrlUtil
+import org.wikipedia.util.StringUtil
+import java.util.Date
+
+class PreviewReadingListRepository(
+    private val context: Context = WikipediaApp.instance,
+    private val prefs: Prefs = Prefs
+) {
+
+    suspend fun getPreviewReadingList(): ReadingList? {
+        val encodedJson = prefs.receiveReadingListsData
```

Comment:
It would be great if use `Prefs.receiveReadingListsData` directly.

Reply:
I've converted it to a variable so as to support constructor injection (and easily write tests later with mocks). What do you think?